datadogfirehosenozzle/datadog_firehose_nozzle.go

```diff
@@ -90,21 +90,34 @@ func (d *DatadogFirehoseNozzle) Start() error {
 	d.db = db
 
 	d.log.Info("Starting DataDog Firehose Nozzle...")
-	d.client = d.createClient()
-	d.cfClient = d.createCfClient()
+	// Create Datadog client instances
+	d.clients, err = d.createClient()
```

Comment:
wow we have two things called client that is not so great lol (that's outside the scope of the PR though)

Reply:
Yeah lot of things could be changed! 
- Starting with replacing datadogclient with dogstatsd or the agent itself.
- The structure of the project is could be improved as well. It is currently very flat where we should make it more like a tree. `worker.go` should be called differently and should be in another package. 
- naming conventions and linting in general
- Creating a specific object class for all things that should be used between threads.
- ...